packages/components/src/Autocomplete/Autocomplete.test.tsx

```diff
@@ -162,3 +162,112 @@ test("it should call the handler skipping headings when an option is selected",
 
   expect(changeHandler).toHaveBeenCalledWith(headingOptions[1].options[0]);
 });
+
+it("should remove the menu when blurred", async () => {
+  const changeHandler = jest.fn();
+  const { getByRole, getByText, queryByText } = render(
+    <Autocomplete
+      value={undefined}
+      onChange={changeHandler}
+      initialOptions={options}
+      getOptions={returnOptions(options)}
+      placeholder="placeholder_name"
+    />,
+  );
+
+  const input = getByRole("textbox");
+
+  input.focus();
+
+  await waitFor(() => {
+    expect(getByText("option_0")).toBeInstanceOf(HTMLParagraphElement);
+  });
+
+  input.blur();
+
+  await waitFor(() => {
+    expect(queryByText("option_0")).toBeNull();
+  });
+});
+
+it("should call onBlur callback when blurred", async () => {
+  const blurHandler = jest.fn();
+  const { getByRole } = render(
+    <Autocomplete
+      value={undefined}
+      onChange={jest.fn()}
+      initialOptions={options}
+      getOptions={returnOptions(options)}
+      placeholder="placeholder_name"
+      onBlur={blurHandler}
+    />,
+  );
+
+  const input = getByRole("textbox");
+  input.focus();
+  input.blur();
+
+  await waitFor(() => {
+    expect(blurHandler).toHaveBeenCalledTimes(1);
+  });
+});
+
+it("should call onChange with undefined if allowFreeForm is false and not matched", async () => {
+  const changeHandler = jest.fn();
+  const { getByRole } = render(
+    <Autocomplete
+      value={undefined}
+      onChange={changeHandler}
+      initialOptions={options}
+      getOptions={returnOptions(options)}
+      placeholder="placeholder_name"
+      allowFreeForm={false}
+    />,
+  );
+
+  const input = getByRole("textbox");
+
+  input.focus();
+
+  fireEvent.input(input, {
+    target: {
+      value: "opt",
+    },
+  });
+
+  input.blur();
+
+  await waitFor(() => {
+    expect(changeHandler).toHaveBeenCalledWith(undefined);
+  });
+});
+
+it("sets the input value to blank if allowFreeForm is false and not matched", async () => {
```

Comment:
Just for my understanding... is it irrelevant to test "sets input to blank when allowFreeForm is true and not matched"?

Reply:
that use case wouldn't be true, with `allowFreeForm` to be true the input can equal a value that is not set in options.

My mindset here was to test everything that I touched, there is probably room for a test around `allowFreeform={true}` for some time in the future